app/change_set_persisters/plum_change_set_persister.rb

```diff
@@ -56,6 +56,7 @@ def before_save(change_set:)
 
     def after_save(change_set:, updated_resource:)
       append(append_id: change_set.append_id, updated_resource: updated_resource) if change_set.append_id.present?
+      IdentifierService.mint_or_update(updated_resource) if change_set.try(:state_changed?) && change_set.new_state == 'complete'
```

Comment:
It works because it saves the resource directly: https://github.com/pulibrary/figgy/pull/95/files#diff-41aae6e0457a6b7d231238cba013d0d1R8

Should we be minting the ARK earlier and getting it in before the change_set is persisted?